MapboxNavigation/RouteMapViewController.swift

```diff
@@ -253,6 +259,66 @@ extension RouteMapViewController: NavigationMapViewDelegate {
         let route = routeController.routeProgress.route
         guard let coordinates = route.coordinates else  { return nil }
         
+        if let userLocation = mapView.userLocation,
+            let style = mapView.style,
+            recenterButton.isHidden {
+            
+            let streetsLanguages = ["zh", "ru", "fr", "es", "en", "de"]
+            let roadLabelLayerIdentifier = "roadLabelLayer"
+            
+            let streetsSources = style.sources.flatMap {
+                $0 as? MGLVectorSource
+                }.filter {
+                    $0.isMapboxStreets
+            }
+            let streetsSourceIdentifiers = streetsSources.map {
+                $0.identifier
+            }
+            assert(!streetsSourceIdentifiers.isEmpty, "Style must contain the source `mapbox.mapbox-streets-v7`")
+            
+            if let mapboxSteetsSource = streetsSources.first, style.layer(withIdentifier: roadLabelLayerIdentifier) == nil{
+                let streetLabelLayer = MGLLineStyleLayer(identifier: roadLabelLayerIdentifier, source: mapboxSteetsSource)
+                streetLabelLayer.sourceLayerIdentifier = "road_label"
+                
+                // If the opacity is set to 0, the feature will be ignored in `mapView.visibleFeatures()`
+                streetLabelLayer.lineOpacity = MGLStyleValue(rawValue: 0.001)
+                streetLabelLayer.lineWidth = MGLStyleValue(rawValue: 10)
+                streetLabelLayer.lineColor = MGLStyleValue(rawValue: .white)
+                style.addLayer(streetLabelLayer)
+            }
+            
+            if let userPuck = mapView.view(for: userLocation),
```

Comment:
All we need for querying is the coordinate (snapped to the route line), not the view itself. Use [`MGLMapView.convert(_:toCoordinateFrom:)`](https://www.mapbox.com/ios-sdk/api/3.5.2/Classes/MGLMapView.html#/c:objc%28cs%29MGLMapView%28im%29convertPoint:toCoordinateFromView:) to get the point corresponding to the snapped location.